test/cpp/jit/test_gpu.cpp

```diff
@@ -18148,6 +18148,135 @@ TEST(NVFuserTest, FusionRfactorContigIDs_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, FusionMMASwizzlePrimitiveFloatAcc_CUDA) {
+  FusionExecutor fe;
+
+  std::string kernel = R"(
+  __global__ void mma32x32x8_mini_gemm_float(Tensor<__half, 2> Ag, Tensor<__half, 2> Bg, Tensor<float, 2> Cg0, Tensor<float, 2> Cg1){
+    // Accumulator
+    float C[32];
+    // initialize C:
+    for(int i=0;i<32;i++){
+        C[i] = 0;
+    }
+    // Allocate Smem
+    __shared__ uint4 As_mem[32*8 / 8];
+    __shared__ uint4 Bs_mem[8*32 / 8];
+    auto As = reinterpret_cast<__half*>(As_mem);
+    auto Bs = reinterpret_cast<__half*>(Bs_mem);
+    // Compute lane_id
+    int lane_id = threadIdx.x % 32;
+    
+    // Mini prolog: (single warp)
+        // global load:
+    __half Ag_buffer[8];
+    *reinterpret_cast<uint4*>(Ag_buffer) = *reinterpret_cast<uint4*>(&Ag[threadIdx.x*8]);
+    __half Bg_buffer[8];
+    *reinterpret_cast<uint4*>(Bg_buffer) = *reinterpret_cast<uint4*>(&Bg[threadIdx.x*8]);
+        // Write into smem: (should consider cleaner utility for vectorization)
+    *reinterpret_cast<uint4*>(
+        &As[
+                // Utility to compose a sequence of swizzles
+            mem_swizzle::swizzle_sequence<
+                    // This is the instruction mem layout
+                    //   for mma32X32X8
+                warp_mma::mmaM32N32K8::SmemWriteASwizzle>(
+                    // The coordinate is what the Ag_buffer data
+                    //   corresponds to in un-swizzled layout
+                    //   mostly compatible with nvfuser indexing
+                    {(int)threadIdx.x, 0}, lane_id).linearize({8,1})])
+    =*reinterpret_cast<uint4*>(Ag_buffer);
+    *reinterpret_cast<uint4*>(
+        &Bs[
+                // Utility to compose a sequence of swizzles
+            mem_swizzle::swizzle_sequence<
+                    // This is the instruction mem layout
+                    //   for mma32X32X8
+                warp_mma::mmaM32N32K8::SmemWriteBSwizzle>(
+                    // The coordinate is what the Bg_buffer data
+                    //   corresponds to in un-swizzled layout
+                    //   mostly compatible with nvfuser indexing
+                    {((int)threadIdx.x*8 / 32), ((int)threadIdx.x*8 % 32)}, lane_id)
+                        .linearize({32,1})])                 
+    =*reinterpret_cast<uint4*>(Bg_buffer);
+    // Read from Smem:
+    __half A[16];
+    __half B[16];
+    // Two 128b reads per operand, this loop to be 
+    //  generated by scheduling
+    for(int warptile_i = 0;warptile_i<2;warptile_i++){
```

Comment:
Is this loop to cover the 32x32x8 tile? The smem loading/storing didn't require any additional loops, should this actually be part of the swizzle ops directly?

This is probably my highest priority question in this PR, as this is the only one that could directly impact the `warp_mma` namespace.

Reply:
This part I'm thinking is exclusively a Volta mma problem. Two smem loads will be needed per instruction tile (32x32x8) because of the data replication. In the generated kernel I just generated 2 loads for these on the fuser side instead of generating a loop.